src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1828,24 +1859,33 @@ impl<T: AstInfo> AstDisplay for WithOptionValue<T> {
 }
 impl_display_t!(WithOptionValue);
 
-impl<T: AstInfo> TryFrom<WithOptionValue<T>> for String {
-    type Error = anyhow::Error;
+macro_rules! try_from_with_option_value_value {
+    ($t:ty) => {
+        impl<T: AstInfo> TryFrom<WithOptionValue<T>> for $t {
+            type Error = anyhow::Error;
 
-    fn try_from(v: WithOptionValue<T>) -> Result<Self, Self::Error> {
-        match v {
-            WithOptionValue::Value(v) => v.try_into(),
-            _ => anyhow::bail!("cannot use value as string"),
+            fn try_from(v: WithOptionValue<T>) -> Result<Self, Self::Error> {
+                match v {
+                    WithOptionValue::Value(v) => v.try_into(),
+                    _ => anyhow::bail!("cannot use value as string"),
```

Comment:
can you inline `$t` here instead of hardcoding "string"? I forget

Reply:
Oh derp--good catch. Will fix in my next push.